core/src/main/java/io/questdb/griffin/engine/functions/bool/AllNotEqStrFunctionFactory.java

```diff
@@ -0,0 +1,89 @@
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2020 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.griffin.engine.functions.bool;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.Function;
+import io.questdb.cairo.sql.Record;
+import io.questdb.griffin.FunctionFactory;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.functions.BooleanFunction;
+import io.questdb.griffin.engine.functions.UnaryFunction;
+import io.questdb.griffin.engine.functions.constants.BooleanConstant;
+import io.questdb.std.CharSequenceHashSet;
+import io.questdb.std.ObjList;
+
+public class AllNotEqStrFunctionFactory implements FunctionFactory {
+
+    @Override
+    public String getSignature() {
+        return "<>all(Ss[])";
+    }
+
+    @Override
+    public Function newInstance(ObjList<Function> args, int position, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) throws SqlException {
+        Function arrayFunction = args.getQuick(1);
+        int arraySize = arrayFunction.getArrayLength();
+        if (arraySize == 0) {
+            return new BooleanConstant(position, true);
+        }
+
+        CharSequenceHashSet set = new CharSequenceHashSet();
+        for (int i = 0; i < arraySize; i++) {
+            set.add(arrayFunction.getStr(null, i));
+        }
+
+        Function var = args.getQuick(0);
+        if (var.isConstant()) {
+            CharSequence str = var.getStr(null);
+            return new BooleanConstant(position, str != null && !set.contains(str));
+        }
+
+        return new AllNotEqualStrFunction(position, var, set);
+    }
+
+    private static class AllNotEqualStrFunction extends BooleanFunction implements UnaryFunction {
+        private final Function arg;
+        private final CharSequenceHashSet set;
+
+        private AllNotEqualStrFunction(int position, Function arg, CharSequenceHashSet set) {
+            super(position);
+            this.arg = arg;
+            this.set = set;
+        }
+
+        @Override
+        public Function getArg() {
+            return arg;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            CharSequence str = arg.getStr(rec);
+            return str != null && !set.contains(str);
```

Comment:
the set supports `null` actually, also there is 
```
set.excludes()
```
just for cases like this :)

There are few other places with similar logic

Reply:
Right, but PostgreSQL returns empty results for queries:

```SQL
select * from tab where null = ALL('{null}');
select * from tab where null <> ALL('{abc}');
```

even though the table `tab` is not empty.

If I understand correctly this behaviour is explained in the documentation:
* https://www.postgresql.org/docs/current/functions-comparisons.html#id-1.5.8.30.17
>If the array expression yields a null array, the result of ALL will be null. If the left-hand expression yields null, the result of ALL is ordinarily null (though a non-strict comparison operator could possibly yield a different result). Also, if the right-hand array contains any null elements and no false comparison result is obtained, the result of ALL will be null, not true (again, assuming a strict comparison operator). This is in accordance with SQL's normal rules for Boolean combinations of null values.
* https://www.postgresql.org/docs/current/functions-comparison.html 
> Ordinary comparison operators yield null (signifying “unknown”), not true or false, when either input is null. For example, 7 = NULL yields null, as does 7 <> NULL

So, in fact for `x <> ALL` I have to always return false when `x` is null.

Reply:
makes sense indeed!